shm/Map.cpp

```diff
@@ -1,6 +1,30 @@
 #include "Map.hpp"
-
 #include <algorithm>
+#include <random>
+#include <iostream>
+
+Map::Map(){
+    islandsLocations_.reserve(amountIslands);
+    generateIslands();
+}
+
+void Map::generateIslands() {
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> distrib(widthIsland, heightIsland);
+    
+      for (size_t i = 0; i < amountIslands; i++) { 
+        size_t positionX; 
+        size_t positionY; 
+        do { 
+            size_t positionX = distrib(gen); 
+            size_t positionY = distrib(gen);
```

Comment:
You have two variables declared twice in two different scopes. You unnecessary re-declared those variables inside do-while loop.